core/shared/src/main/scala/fs2/Compiler.scala

```diff
@@ -34,31 +34,29 @@ import scala.annotation.implicitNotFound
   "Cannot find an implicit Compiler[F, G]. This typically means you need a Concurrent[F] in scope"
 )
 sealed trait Compiler[F[_], G[_]] {
-  private[fs2] def apply[O, B, C](stream: Pull[F, O, Unit], init: () => B)(
-      fold: (B, Chunk[O]) => B,
-      finalize: B => C
-  ): G[C]
+  private[fs2] val target: Monad[G]
```

Comment:
Alternatively, we could change `.to(..)` to take a `Monad[G]` constraint, which would then percolate up to `toList`, `toVector`, and `string`.